scripts/main.js

```diff
@@ -1271,6 +1275,40 @@
                 break;
         }
 
+        // Set a value for LNY bonus luck, if it can be determined.
+        let quest = response.user.quests.QuestLunarNewYear2019;
+        if (quest) {
+            // Avoid overwriting any existing hunt details.
+            if (!message.hunt_details) {
+                message.hunt_details = {};
+            }
+            message.hunt_details.is_lny_hunt = true;
+
+            let lny_luck = null;
+            if (quest.lantern_status.includes("noLantern")) {
+                // The user has no pig lantern yet, so they cannot have bonus LNY luck.
+                lny_luck = 0;
+            } else if (quest.is_lantern_active) {
+                // Each hunt with the active lantern increases the height by 1, and thus possibly the luck.
+                lny_luck = getLNYLuck(quest);
+            } else {
+                // The lantern is inactive. This may be an explicit choice, or because the user ran out of equipped candles.
+                let yellow = quest.items.lny_unlit_lantern_stat_item;
+                let red = quest.items.lny_unlit_lantern_2018_stat_item;
+                // If the user has both candles, we know the lantern state was an explicit
+                // choice. (Candles given via height rewards must be explicitly claimed.)
+                if (parseInt(yellow.quantity, 10) > 0 && parseInt(red.quantity, 10) > 0) {
+                    lny_luck = 0;
+                } else if (yellow.status === "active" || red.status === "active") {
```

Comment:
The idea right here is that maybe one of the candles still reports if it was active even if the player ran out of it on the hunt. I'll be able to determine how the game responds in a few hours.